mlflow/tracking/__init__.py

```diff
@@ -1,413 +1,19 @@
-    """
-    if not isinstance(value, numbers.Number):
-        print("WARNING: The metric {}={} was not logged because the value is not a number.".format(
-            key, value), file=sys.stderr)
-        return
-    _get_or_start_run().log_metric(Metric(key, value, int(time.time())))
-
-
-def log_artifact(local_path, artifact_path=None):
-    """Log a local file or directory as an artifact of the currently active run."""
-    _get_or_start_run().log_artifact(local_path, artifact_path)
-
-
-def log_artifacts(local_dir, artifact_path=None):
-    """Log all the contents of a local directory as artifacts of the run."""
-    _get_or_start_run().log_artifacts(local_dir, artifact_path)
-
-
-def get_artifact_uri():
-    """
-    Return the artifact URI of the currently active run. Calls to ``log_artifact`` and
-    ``log_artifacts`` write artifact(s) to subdirectories of the returned URI.
-    """
-    return _get_or_start_run().get_artifact_uri()
-
-
-atexit.register(end_run)
-
-
-def _get_model_log_dir(model_name, run_id):
-    if not run_id:
-        raise Exception("Must specify a run_id to get logging directory for a model.")
-    store = _get_store()
-    run = store.get_run(run_id)
-    if run.info.artifact_uri:
-        artifact_repo = ArtifactRepository.from_artifact_uri(run.info.artifact_uri, store)
-    else:
-        artifact_repo = _get_legacy_artifact_repo(store, run.info)
-    return artifact_repo.download_artifacts(model_name)
-
-
-def _get_legacy_artifact_repo(file_store, run_info):
-    # TODO(aaron) Remove this once everyone locally only has runs from after
-    # the introduction of "artifact_uri".
-    uri = os.path.join(file_store.root_directory, str(run_info.experiment_id),
-                       run_info.run_uuid, "artifacts")
-    return ArtifactRepository.from_artifact_uri(uri, file_store)
-
-
-def _get_git_commit(path):
-    try:
-        from git import Repo, InvalidGitRepositoryError, GitCommandNotFound, NoSuchPathError
-    except ImportError as e:
-        print("Notice: failed to import Git (the git executable is probably not on your PATH),"
-              " so Git SHA is not available. Error: %s" % e, file=sys.stderr)
-        return None
-    try:
-        if os.path.isfile(path):
-            path = os.path.dirname(path)
-        repo = Repo(path, search_parent_directories=True)
-        commit = repo.head.commit.hexsha
-        return commit
-    except (InvalidGitRepositoryError, GitCommandNotFound, ValueError, NoSuchPathError):
-        return None
+from mlflow.tracking.api import MlflowService, get_service
+from mlflow.tracking.utils import set_tracking_uri, get_tracking_uri, _get_model_log_dir, \
+    _get_store, _is_local_uri, _is_databricks_uri, _TRACKING_URI_ENV_VAR
+from mlflow.tracking.flow import _EXPERIMENT_ID_ENV_VAR, _RUN_ID_ENV_VAR, _get_git_commit
+
+__all__ = [
+    "MlflowService",
+    "get_service",
+    "get_tracking_uri",
+    "set_tracking_uri",
+    "_get_git_commit",
```

Comment:
Q: why expose these internal methods in `__all__`? Doing so means that `from mlflow.tracking import *` will import methods like `_get_git_commit`.

If `_get_git_commit` etc is used in other modules maybe we could make it a "public" API within `mlflow.tracking.flow` (e.g. `mlflow.tracking.flow.get_git_commit`), since `mlflow.tracking.flow` itself won't be a publicly documented module (assuming I understand correctly). Admittedly this would make our code less robust to future refactors of `mlflow.tracking.flow`.